app.js

```diff
@@ -26,10 +66,17 @@ app.use(function(req, res, next) {
 });
 
 app.get('/validate', function (req, res, next) {
-  const token = firebase.auth().createCustomToken(req.query.token);
-  res.json({ token });
+  var apiToken = req.query.token;
+  if (isValidToken(apiToken)) {
+    getApiUser(apiToken).then(function() {
```

Comment:
I don't think it responds with anything - there should be a `catch` here for when `getApiUser` throws an error